### ISSUE TITLE
Error in equivalent implementation - scaled dot product attention

### ISSUE THREAD

**Issue by tkz789** [pytorch/pytorch#117939]
### 📚 The doc issue

https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html

In the equivalent description provided on the website a mask is not applied if dtype of the mask is torch.bool

### Suggest a potential alternative/fix

I suppose that the mask should be added to bias after converting to -inf .
